infrasec/tutorials/atlantis.md

```diff
@@ -0,0 +1,418 @@
+
+1. Create the Atlantis role and policy
+
+    Our role, policy, and policy attachment will look a lot like [the example in legendary waddle](https://github.com/trussworks/legendary-waddle/blob/master/trussworks-prod/atlantis-global/main.tf#L22-L48). We'll want to alter the policy principals to match our project's role names, but otherwise the code is pretty standard. Depending on our logs setup, we may want to add access to service logs as an additional principal like so:
+
+    ```hcl
+    principals {
+      type = "Service"
+      identifiers = [
+        "delivery.logs.amazonaws.com"
+      ]
+    }
+    ```
+
+    We'll also need to add a line of code to let Atlantis role control terraform. In `legendary-waddle` this is done per account by adding a single line of code to the [`role_arn` value to the s3 backend](https://github.com/trussworks/legendary-waddle/blob/master/trussworks-misty/atlantis-global/terraform.tf#L10), and an [`assume_role` object to the account provider](https://github.com/trussworks/legendary-waddle/blob/d896c9efb00bf2fb6ca0bf883747852d1851840b/trussworks-misty/admin-global/providers.tf#L3-L5). We may want to begin by leaving this code commented out. See the [IAM troubleshooting section](#general-iam-role-assumptions-troubleshooting) for caveats regarding role assumption.
+
+2. Create the ECS policy for the Auto-Created Role
+
+    The Atlantis module creates the ECS task automatically (using your name variable and called `<NAME>-ecs_task_execution`) by passing in the [`terraform-aws-ecs`](https://registry.terraform.io/modules/terraform-aws-modules/ecs/aws/latest) module as a submodule, which in turn uses the [aws_iam_instance_profile](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/iam_instance_profile) resource to create the IAM role for the task. As you can see, we're now a few layers deep, which makes any variance in the expected chain of events a bit tricky to troubleshoot.
+
+    For our example, we'll need to create:
+
+    * a [policy](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/iam_policy)
+    * a [policy document](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/iam_policy_document), and
+    * a [role policy attachment](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/iam_role_policy_attachment)
+
+    Our final code will look look something like this:
+
+    ```hcl
+    resource "aws_iam_policy" "atlantis" {
+      name   = "atlantis"
+      path   = "/"
+      policy = data.aws_iam_policy_document.atlantis.json
+    }
+
+    data "aws_iam_policy_document" "atlantis" {
+      # Atlantis can read information about clusters, services, tasks, and task definitions.
+      statement {
+        actions   = ["ssm:*"]
+        effect    = "Allow"
+        resources = ["*"]
+      }
```

Comment:
```suggestion
      statement {
        actions   = ["ssm:GetParameters"]
        effect    = "Allow"
        resources = ["*"] # This can be restricted to the particular SSM parameters Atlantis uses
      }
```
Suggested change to limit the SSM permissions.